### PR TITLE
hle: Eliminate need to specify command headers for IPC.

### PR DESCRIPTION
**Pull request by Steveice10**
* RequestParser now uses the header from the command buffer to parse the request instead of requiring the ID and parameter counts to be passed in manually.
* FunctionInfo now only requires the command ID instead of the full command header. There is never a case where we need to match on the full header; command IDs are always unique for a service.
* Removes specialized templates in CFG and NWM services for commands reused with multiple IDs, since they are no longer necessary with these changes.

Did some basic tests with the home menu and some games, everything seems to work as expected. Most of the changes here are just find-and-replace removing parameters from parsers and pulling the IDs out of function table headers.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6678)

### CODE REVIEW

File: src/core/hle/kernel/hle_ipc.h [hunk 1]
```diff
@@ -207,6 +207,11 @@ class HLERequestContext : public std::enable_shared_from_this<HLERequestContext>
         return cmd_buf.data();
     }
 
+    // Returns the command header from the IPC command buffer.
```
Comment (SachinVin) [inline src/core/hle/kernel/hle_ipc.h:210]:
```suggestion
    /// Returns the command header from the IPC command buffer.
```

**Comment by SachinVin** [comment]
I feel this make it a little harder to cross-reference functions with 3dbrew

**Comment by Steveice10** [comment]
> I feel this makes it a little harder to cross-reference functions with 3dbrew

Any command on 3dbrew you can take the first four header digits and it’s the command ID, which will be unique within each service.

**Comment by SachinVin** [comment]
Alright got it

File: src/core/hle/kernel/hle_ipc.h [hunk 2]
```diff
@@ -207,6 +207,11 @@ class HLERequestContext : public std::enable_shared_from_this<HLERequestContext>
         return cmd_buf.data();
     }
 
+    /// Returns the command header from the IPC command buffer.
+    IPC::Header CommandHeader() {
```
Comment (GPUCode) [inline src/core/hle/kernel/hle_ipc.h:211]:
```suggestion
    IPC::Header CommandHeader() const {
```

**Review by GPUCode — APPROVED** [review]
LGTM. Real services only check command id too, so this should be more accurate.

**Pull request merged**